onnxruntime/core/codegen/mti/nn/pool_ops.cc

```diff
@@ -3,55 +3,60 @@
 
 #include "core/codegen/mti/nn/pool_ops.h"
 
+#include "core/codegen/mti/mti_tvm_utils.h"
+#include "core/mlas/inc/mlas.h"
+#include "core/providers/cpu/nn/pool_attributes.h"
 #include <topi/nn/pooling.h>
 
 namespace onnxruntime {
 namespace tvm_codegen {
 
 // TODO: topi only support 2d-pool, MaxPool1d and MaxPool3d will need to be added if necessary.
 // only support version < 8 for topi doesn't come with implementation to output index tensor
-tvm::Tensor MaxPool(
-    const tvm::Tensor& input,
-    const tvm::Array<tvm::Expr>& kernel_size,
-    const tvm::Array<tvm::Expr>& stride_size,
-    const tvm::Array<tvm::Expr>& padding_size,
-    const std::string& layout,
-    bool count_include_pad) {
-  return topi::nn::pool(input, kernel_size, stride_size, padding_size,
-                        topi::nn::kMaxPool,
-                        false,
-                        layout,
-                        count_include_pad);
+tvm::Tensor MaxPool(const tvm::Tensor& input,
+                    const PoolAttributes& pool_attrs,
+                    const tvm::Array<tvm::Expr>& /*output_shape*/,
+                    const std::string& /*name*/) {
+  return topi::nn::pool(input,
+                        ToTvmArray(pool_attrs.kernel_shape),
+                        ToTvmArray(pool_attrs.strides),
+                        ToTvmArray(pool_attrs.pads),
+                        /*pool_type*/ topi::nn::kMaxPool,
+                        /*ceil_mode*/ false,
+                        /*layout*/ pool_attrs.storage_order == 0 ? "NCWH" : "NCHW",
```

Comment:
>NCWH [](start = 68, length = 4)

This is strange, usually it's either NCHW or NHWC...

Reply:
This is how the existing mti pool ops are implemented:

https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/codegen/passes/op_ir_creator/nn/pool_ops.cc#L40

https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/codegen/passes/op_ir_creator/nn/pool_ops.cc#L48

Reply:
The spec shows storage order is for column major vs. row major thus I have some doubt. Please make sure different storage order is covered in unit test, and as long as those tests passed I'm fine with it.

Reply:
storage_order attribute is always accompanied with optional output Indices, which we don't support yet in Nuphar. So, although there are some unit tests for testing different storage orders, they don't apply to Nuphar providers. We should be able to run those tests once we support Indices.

BTW, by looking at CPU's implementation, seems NCHW and NCWH are the correct ones:

https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/cpu/nn/pool.cc#L335